### PR TITLE
Persist recipes

### PR DESCRIPTION
**Pull request by Tim-Linschoten**
This change persist the recipes separately instead of always use the latest in memory option.
Also changes the way we support multi recipe from using RecipeHandlers to using a recipeId.
This is a smaller change and easier to use.

### CODE REVIEW

File: recipe-dsl/src/main/scala/com/ing/baker/recipe/javadsl/util.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.ing.baker.recipe.javadsl
+
+import com.ing.baker.recipe.common
+
+object util {
```
Comment (merlijn) [inline recipe-dsl/src/main/scala/com/ing/baker/recipe/javadsl/util.scala]: perhaps use capital U -> Util

File: runtime/src/main/scala/com/ing/baker/runtime/actor/ClusterBakerActorProvider.scala [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    s"index-${Math.abs(sha256HashCode(processId) % nrOfShards)}"
+
+  // extracts the actor id -> message from the incoming message
+  // Entity id is the first character of the UUID
+  def entityIdExtractor(nrOfShards: Int): ExtractEntityId = {
+    case msg@CreateProcess(_, processId) => (entityId(processId, nrOfShards), msg)
+    case msg@HandleEvent(processId, _) => (entityId(processId, nrOfShards), msg)
+    case msg@GetProcessState(processId) => (entityId(processId, nrOfShards), msg)
+    case msg@GetCompiledRecipe(processId) => (entityId(processId, nrOfShards), msg)
+    //TODO decide how to shard now without the BakerActorMessage
```
Comment (merlijn) [inline runtime/src/main/scala/com/ing/baker/runtime/actor/ClusterBakerActorProvider.scala]: This TODO is still relevant?

File: runtime/src/main/scala/com/ing/baker/runtime/actor/ClusterBakerActorProvider.scala [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    //    case msg@BakerActorMessage(processId, _) => (entityId(recipeName, processId, nrOfShards), msg)
+  }
+
+  // extracts the shard id from the incoming message
+  def shardIdExtractor(nrOfShards: Int): ExtractShardId = {
+    //TODO decide how to shard now without the BakerActorMessage
```
Comment (merlijn) [inline runtime/src/main/scala/com/ing/baker/runtime/actor/ClusterBakerActorProvider.scala]: This TODO is still relevant?

File: runtime/src/main/scala/com/ing/baker/runtime/java_api/JBaker.scala [hunk 4]
```diff
@@ -48,11 +50,260 @@ class JBaker(actorSystem: ActorSystem, implementations: java.lang.Iterable[AnyRe
...
     * @return
     */
-  def getRecipeHandler(name: String): JRecipeHandler = new JRecipeHandler(baker.getRecipeHandler(name))(actorSystem)
+  def getEvents(processId: UUID): EventList = getEvents(processId.toString, defaultTimeout)
+
+  /**
+    * Returns the compiled recipe.
+    *
+    * @return The compiled recipe
+    */
```
Comment (merlijn) [inline runtime/src/main/scala/com/ing/baker/runtime/java_api/JBaker.scala]: Could throw a TimeoutException, could add something like: 

```@throws[TimeoutException]("When baker does not respond within the default deadline")```

Same for some other methods.

File: recipe-dsl/src/main/scala/com/ing/baker/recipe/javadsl/util.scala (hunk 1 above)
Comment (Tim-Linschoten) [reply recipe-dsl/src/main/scala/com/ing/baker/recipe/javadsl/util.scala]: Done

File: runtime/src/main/scala/com/ing/baker/runtime/actor/ClusterBakerActorProvider.scala (hunk 2 above)
Comment (Tim-Linschoten) [reply runtime/src/main/scala/com/ing/baker/runtime/actor/ClusterBakerActorProvider.scala]: Nope, removed

File: runtime/src/main/scala/com/ing/baker/runtime/actor/ClusterBakerActorProvider.scala (hunk 3 above)
Comment (Tim-Linschoten) [reply runtime/src/main/scala/com/ing/baker/runtime/actor/ClusterBakerActorProvider.scala]: Nope, removed

File: runtime/src/main/scala/com/ing/baker/runtime/java_api/JBaker.scala (hunk 4 above)
Comment (Tim-Linschoten) [reply runtime/src/main/scala/com/ing/baker/runtime/java_api/JBaker.scala]: Will do this in the other branch where we already refactor the timeout part for Baker/JBaker.

**Pull request merged**